Explorer/Assets/DCL/SDKComponents/Tween/Components/CustomTweener/CustomTweener.cs

```diff
@@ -0,0 +1,67 @@
+﻿using CRDT;
+using DCL.ECSComponents;
+using DG.Tweening;
+using DG.Tweening.Core;
+using DG.Tweening.Plugins.Options;
+using UnityEngine;
+
+namespace DCL.SDKComponents.Tween.Components
+{
+    public abstract class CustomTweener<T, TU> : ICustomTweener
+        where T : struct
+        where TU : struct, IPlugOptions
+    {
+        protected TweenerCore<T, T, TU> core;
+        protected T CurrentValue { get;  set; }
+
+        protected Transform StartTransform;
+        private bool Finished;
+
+        public void Play()
+        {
+            core.Play();
+        }
+
+        public void Pause()
+        {
+            core.Pause();
+        }
+
+        public void Rewind()
+        {
+            core.Rewind();
+        }
+
+        public bool IsPaused()
+        {
+            return !core.IsPlaying();
+        }
+
+        public bool IsFinished()
+        {
+            return Finished;
+        }
+
+        public bool IsActive()
+        {
+            return !core.IsPlaying() && !Finished;
+        }
+
+        public void DoTween(Ease ease, float tweenModelCurrentTime, bool isPlaying)
+        {
+            core.SetEase(ease).SetAutoKill(false).OnComplete(() => { Finished = true; }).Goto(tweenModelCurrentTime, isPlaying);
+        }
+
+        public void Dispose()
+        {
+            core?.Kill();
+        }
+
+        protected abstract TweenerCore<T, T, TU> CreateTweener(T start, T end, float duration);
+
+        public abstract TweenResult GetResult();
```

Comment:
I propose you to write current (default) values to `TweenResult` in the base class (non-virtual method, mandatory) and then modify the desired component only in the derived classes (in the abstract method) as it's less error-prone and will remove the same logic duplication in every case